buildSrc/src/main/kotlin/com.mongodb.intellij.plugin-component.gradle.kts

```diff
@@ -120,12 +120,14 @@ jmhReport {
 }
 
 tasks {
-    register("buildProperties", WriteProperties::class) {
-        group = "build"
+    if (pluginBundle.enableBundle.get() == true) {
+        register("buildProperties", WriteProperties::class) {
+            group = "build"
 
-        destinationFile.set(project.layout.projectDirectory.file("src/main/resources/build.properties"))
-        property("pluginVersion", rootProject.version)
-        property("segmentApiKey", System.getenv("BUILD_SEGMENT_API_KEY") ?: "<none>")
+            destinationFile.set(project.layout.projectDirectory.file("src/main/resources/build.properties"))
+            property("pluginVersion", rootProject.version)
+            property("segmentApiKey", System.getenv("BUILD_SEGMENT_API_KEY") ?: "<none>")
```

Comment:
Should we fail the build here - otherwise if we misconfigure the build environment, we can end up with a bundle without a segment key.

Reply:
True, that is exactly what happened tbf when we modernised the build configuration with jetbrains best practices.

The thing is that we want to be able to build a version without access to segment. Ideally, we will enable Segment only in "main" and disable segment in the "beta" channel. We need to think more about this.

Reply:
Perhaps we fail the build if that environment variable is not present and we set it to "<none>" for beta/local builds?